src/util/GameLogicUtils.hpp

```diff
@@ -76,6 +76,37 @@ namespace spy::util {
             static std::vector<character::Character>::iterator
             getInCharacterSetByCoordinates(character::CharacterSet &cs, const util::Point &p);
 
+            /**
+             * Returns a random point on the map for which the predicate returns true
+             * @tparam Predicate function-like type mapping spy::util::Point -> bool
+             * @param s current game state
+             * @param p predicate indicating valid fields, gets called exactly once for each field on the map
+             * @return random point for which p returns true, std::nullopt if p returns false for all points
+             */
+            template<typename Predicate>
+            static std::optional<util::Point> getRandomMapPoint(const gameplay::State &s, Predicate p) {
+                static std::mt19937 rng{std::random_device{}()};
+
+                // List of all fields matching predicate
+                std::vector<util::Point> validFields{};
+                for (auto row = 0U; row < s.getMap().getNumberOfRows(); row++) {
+                    for (auto col = 0U; col < s.getMap().getRowLength(row); col++) {
+                        auto point = util::Point{static_cast<int>(col), static_cast<int>(row)};
+                        if (p(point)) {
+                            validFields.push_back(point);
+                        }
+                    }
+                }
```

Comment:
hier könnte man auch die `getAllFieldsWith` Funktion aus GameLogicUtils verwenden

Reply:
fixed it